### DISCUSSION TITLE
Please help. I Can't connect Xdrip, Can't scan QR code

### DISCUSSION THREAD

**Discussion by iNgobe** [NightscoutFoundation/xDrip#3655] [Q&A] [answered]
Please help me get Xdrip set up and connected.
  I have a Samsung S23Ultra (Android 14) which I've used to connect Xdrip via the Dexcom app since G7's became available in UK. This worked until about a month ago when it irreparably lost connection.  My control is suffering while I rely on Dexcom Clarity alone & I desperately need Xdrip+ working again. 
  I have uninstalled Dexcom & tried repeatedly installing Xdrip+ latest release alpha version, tried reinstalling beta version, tried restarting phone, clearing cache, but nothing works. I have enabled location, and allowing Xdrip to run in background without battery optimisation, I've tried reinstalling old settings, tried not reinstalling old settings in case that included incorrect settings of old setup as Dexcom follower which died. 
  No matter which version I install very time I cannot scan the Qr code as the Xdrip scanner just doesn't scan, and I also can't access a saved scan image in my gallery. 
  If I try to pair with sensor, it only sometimes sees the sensor but then fails to pair, sometimes because Xdrip 'has a bug and closed' or because I haven't scanned QR code, which I cannot do but no alternative options appear such as input code manually (although it does have this sensor code correctly listed in settings  already as it was input in the source wizard). 
Please help with any advice for a non-coder ignoramus who needs simple instructions as I am lost without Xdrip.

**Comment by iNgobe** [comment]
Just to clarify, I am trying to connect a fresh installation of Xdrip to an already inserted G7 sensor. Attached 
![Screenshot_20240905_150327_xDrip+](https://github.com/user-attachments/assets/0dc25fb7-117a-46ae-be82-6ac3569f3f91)
is a screenshot of the current system status

**Comment by Navid200** [comment]
There is no point in doing anything else if you cannot scan the QR code.  Changing the settings is not going to let you get readings without the QR code.

Install this version of xDrip: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.08.02  

Download the QR code from the following page onto your phone?
https://navid200.github.io/xDrip/docs/Dexcom/G7.html  

Then, follow the instructions under "If camera is not an option" here to scan the QR code you saved to your Download folder:
https://navid200.github.io/xDrip/docs/Settings/Scan_QR_Code.html

**Comment by Navid200** [comment]
If you need Clarity, you need to get Dexcom app going: https://navid200.github.io/xDrip/docs/Dexcom/Clarity.html  

If you cannot get Dexcom app working on your phone, considering your phone is on the Dexcom compatibility list, you should contact Dexcom and figure out what the cause of the problem is.
xDrip is not going to be your solution.  
xDrip is not an official app.  
Something is obviously wrong with your phone or how it is set up and you should fix that first.

**Reply by iNgobe** [reply]
Thanks for your reply.

I don't need Clarity. I had to have it installed as Xdrip initially couldn't get readings directly from G7 & had to connect via Dexcom. As that was working for me I didn't even know that Xdrip now connects directly to sensor. Instructions I saw here advised to install Dexcom so I did. 

I don't know what the issue with Xdrip not scanning the QR code is but it seems to be within Xdrip as my phone camera works fine and scans QR codes fine, and I have enabled camera permissions for Xdrip.

I will try installing the link you gave above.

**Comment by iNgobe** [comment]
Thanks again. 
So far following your advice I have achieved PAIRING, but not yet connection.
![Screenshot_20240905_155107_xDrip+](https://github.com/user-attachments/assets/1e9a0194-bc77-4f40-8729-56d01e192bf1)

**Comment by iNgobe** [comment]
Fantastic! I'm connected at last!
Thank you.
I'll look into whether my phone has camera issues. What should appear onscreen when Xdrip QR code scanner is working correctly?
I get a long rectangle so have to turn my phone to landscape, with area outside it blurred and a red line  hovering across the middle of the rectangle. There is no response to touching the screen, no movable or fixed frame marks and below the screen text says something about click or press or just SCAN the code, but there is nothing to click on or initiate a scan.

**Reply by Navid200** [reply]
It is a known xDrip issue.  Hopefully, we will resolve it.  But, it varies from phone to phone and only happens on recent Android versions.  So, it could take a while.
That's why we added an alternative way for scanning the QR code.  There are even phones that have no camera.  Or the camera may be broken.  With the option to scan from a file, this is not really a critical issue any longer.

**Reply by Navid200** [reply]
This is what it should look like where you can see a scanning window in the center of your screen:  
![Screenshot_20240905-113126](https://github.com/user-attachments/assets/7a0febb3-db04-42f9-a7d2-70ff9acada8a)  
<br/>  
  
And this is what it looks like when there is the issue resulting in the scanning window to move off center:  
![1](https://github.com/user-attachments/assets/8db0393e-7bf7-4930-be1d-d49527100be4)  

There are open issues and it is a very small group of us.  I am trying to close the high priority issues first.  We will eventually get to this I hope.

**Reply by iNgobe** [reply]
I do get the scanning window in the centre of the screen exactly as in your first pic, 
but what I don't know how to do is how to 'action' or start or take the scan? 
 Is the red horizontal line meant to be held across the centre of the QR code or where? Holding the camera anywhere over the QR code doesn't result in an automatic scan for me and there is no button to click or action for me to take to cause a scan to be taken or saved.
 Should there be an action button or does the scanner automatically capture a scan if it is working properly?

**Reply by Navid200** [reply]
The action is to press auto configure in xDrip.  After that when the scanner window opens, it will scan if it sees a QR code.  You should rotate the phone so that it is in landscape more.  You should hold the phone such the the red line goes over the QR code.  You need to move the phone back so that the camera can focus.

As explained here: https://navid200.github.io/xDrip/docs/Settings/Scan_QR_Code.html

**Comment by iNgobe** [comment]
I did press auto-configure, I did turn the open scanner window to landscape mode, I did hold the camera close, I did hold the camera a medium distance away, and I did move the camera further away so the QR code was in focus, I did try holding the red line centrally in the middle of the QR code, I did try holding the red line at the bottom of the QR code, and I did try slowly moving the red line from the base up to the top of the QR code, but no matter where I held it no scan was captured at all - so that is why I asked is there an action I must take or a button I should click to tell it to take a scan, because it simply does not do it automatically.  Should it do it automatically and exactly where within the QR code must the red line be placed?

**Reply by Navid200** [reply, marked as answer]
No, there is no additional action required when it works.

**Reply by iNgobe** [reply]
Thanks again for your help Navid200. Just to inform you that regarding Xdrip QR scanning issues using Camera on Samsung phones, the first Samsung advisor I asked agreed with you & said my S23ULTRA camera isn't working as it shows links from QR codes but doesn't automatically open them (so it doesn't work for scanning code within Xdrip), & told me to try phone reset to default. This didn't work but the next Samsung advisor said this advice was incorrect and no Samsung phone Camera app automatically opens QR codes, instead they offer a link to manually click on to open it. Direct automatic opening of QR codes can only be done using Samsung QR code reader which is hidden in settings, but not using the Samsung Camera app. Not sure who is actually right but fortunately I have got X-drip running without scanning.
